### PR TITLE
release: `v0.29.0`

### LINKED ISSUE #1263: Release an 0.29 with #1245 included
### Description

Hi, it'd be great to get a release of `tendermint-rs` with #1245 included. Its exclusion from the `tendermint-rs` 0.34 backport regressed our debug output, causing a lot of debug spam for the last few months, and we've been hoping to avoid having to either rewrite all of our debug output or re-fork the dependency.

### Definition of "done"

A version of `tendermint-rs` with #1245 included is released and propagated through the rest of Informal's dep tree (particularly `ibc-rs`).

### PR DESCRIPTION
**Pull request by erwanor**
Hey there, this fixes #1263  :)

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Added entry in `.changelog/`

### CODE REVIEW

**Review by thanethomson — CHANGES_REQUESTED** [review]
Thanks very much for this @erwanor!

There are a couple of places on this branch where the `html_root_url` version hasn't been updated to v0.29.0:
- `tendermint/src/lib.rs`
- `light-client/src/lib.rs`
- `p2p/src/lib.rs`

Also, note that pushing a v0.29.0 tag will not yet automatically build and push the crates to crates.io - this is still quite a manual process that involves running the `release.sh` script from the root of the repo.

We still need a GitHub workflow to do this (please speak to @hdevalence about this, as I shared details a few days ago about what we need there).

**Comment by erwanor** [comment]
Thanks for catching those, I'll update the docs!

**Comment by thanethomson** [comment]
Ah, and I just noticed you're missing the `.changelog/v0.29.0` folder.

**Comment by tarcieri** [comment]
Note: if you’re using docs.rs, setting `html_root_url` manually is obsolete: https://github.com/rust-lang/api-guidelines/issues/229

**Comment by thanethomson** [comment]
> re: `release.sh` do I have the appropriate permissions to push to crates.io once this PR gets approved?

No, I've just pushed #1268 to attempt to help with this. We've configured our @bftbot to be able to push crates on our behalf when you push tags to the repo. Please double-check my approach in #1268 to see if I've missed anything important?

If we can merge #1268 first, then this PR, then if you push the `v0.29.0` tag to the repo it should trigger the release. Please also sign and annotate the tag before pushing it:

```bash
git tag -s -a v0.29.0 -m 'Release v0.29.0'
```

I usually manually create a release from a new tag. Eventually we'll automate that too.

> Note: if you’re using docs.rs, setting `html_root_url` manually is obsolete: [rust-lang/api-guidelines#229](https://github.com/rust-lang/api-guidelines/issues/229)

Thanks for the heads-up @tarcieri!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -183,8 +183,6 @@ Our release process is as follows:
    changes to the release PR. This includes:
    1. All `Cargo.toml` files (making sure dependencies' versions are updated
       too).
-   2. All crates' `lib.rs` files documentation references' `html_root_url`
```
Comment (thanethomson) [inline CONTRIBUTING.md:186]: Please edit point (3) above to make it a single sentence.

File: .changelog/v0.29.0/summary.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+*Feb 17, 2023*
+
+This release features modularity improvements for the cryptographic routines, as well as fixes related to block verification and the use of consensus-friendly ed25519 crate.
```
Comment (thanethomson) [inline .changelog/v0.29.0/summary.md:3]:
```suggestion
This release features modularity improvements for the cryptographic routines, as well as fixes related to block verification and the use of a consensus-friendly ed25519 crate.
```

Comment (thanethomson) [reply .changelog/v0.29.0/summary.md:3]: To match the generated changelog.

**Review by thanethomson — APPROVED** [review]
🎉

**Pull request merged**